add musl docker image build

clux opened the pull request:
A slightly different approach to #74 than #77 for a smaller image.

Uses [muslrust](https://github.com/clux/muslrust) for cross compiling for musl, and adds the static binaries into an alpine image:

```
REPOSITORY                      TAG                                        IMAGE ID            CREATED             SIZE
luminarys/synapse               latest                                     de4848aebac6        9 minutes ago       17.8MB
```

Think it handles download volumes and ports sufficiently? crt/pem files can be mounted over in the users homedir. Any other config changes can be done by inheriting from this image. Feels like sensible defaults to me.

The rest is my normal makefile template for these things. Trash as appropriate.

Side-discovery: docker needs a lowercase name in the organisation.

clux committed:
add musl docker image build

Luminarys commented:
Thanks! This seems more in line with what I'd like for a container.

Luminarys commented:
The example config will need to be tweaked to listen non-locally.

clux committed:
listen non-locally in docker container

Luminarys commented:
Additionally, the mounted volume should be chown'd to synapse, or else downloading won't work

Luminarys commented:
Also using a .dockerignore along the lines of:
```
*
!target/x86_64-unknown-linux-musl/release/synapse
!target/x86_64-unknown-linux-musl/release/sycli
!example_config.toml
```
would be good

clux committed:
dockerignore

clux committed:
chown downloads dir to user inside container

if your $(id -u):$(id -g) == 1000:1000 (standard linux one user) then you don't need to chown the
files touched outside the container. Otherwise chowing outside might be
needed occasionally.

clux commented:
There, forgotten how to do this sanely for non-root containers. It's still a little iffy if your main user isn't 1000:1000, but people can `sudo cp` out from the volume mount and `chown` the files I guess.

Luminarys commented:
Would it not make more sense to do `docker run -u $UID` and skip the `chown` bit? That and why is the run command now `sh`?

clux commented:
Well it's not really sufficient to just set `-u`. If the id's are not the same inside and outside there's often issues. Usually most people run 1000:1000 so we can skip the `chown` for the majority case - in this case we don't even need `-u` (because it's implicitly going to use 1000:1000 inside due to the USER instruction).

However, running this on a dummy user on my system with 1002:1002, then no variation of `-u` would work consistently without a corresponding `chown`.

clux committed:
remove accidental sh

Luminarys commented:
Thanks!
